Remove TINC storage.persistent_tables tests.

jimmyyih opened the pull request:
These tests do not really give us any extra coverage different from
our other TINC catalog and storage tests. Historically, they haven't
flagged any persistent table issues we have seen in the field and we
were surprised it didn't flag any persistent table issues from the 8.3
merge but other TINC tests did. Therefore, it is safe to remove these
tests without fear of losing coverage.

ashwinstar commented:
Thank You 👍 Just would be worth for history to include your great summary from tracker of what it tested in commit message.

cjcjameson commented:
Sweet. Yes, I'm curious too.

jimmyyih commented:
Updated commit to leave PT rebuild test alone (@Chibin will take care of this one when they migrate it to Behave) and added brief summary of tests that are being removed to commit message.
